src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<Bulldozer> bulldozers = new BulldozerProducer().get();
```

Comment:
```suggestion
            return new ArrayList<>(bulldozerProducer.get());
```
lets make producers class-level variables, because you are creating a new producer each time this method is invoked
ps. choose abstract type for the variable